gnmpagerduty/tasks.py

```diff
@@ -0,0 +1,238 @@
+    :param val: metadata returned from storage_data map about the storage
+    :return: None
+    """
+    if key.incident_key == '':
+        #if no incident has yet been registered, send one to PD
+        percent_free = return_percentage(int(val['capacity']), int(val['freeCapacity']))
+
+        pddata = notify_pagerduty("Storage {0} lacks sufficient free space. It is {1}% full.".format(val['contentDict']['name'],percent_free),
+                         "trigger",
+                         key.incident_key,
+                         vidis_id=val['name'],
+                         storage_name=val['contentDict']['name'],
+                         free_capacity=human_friendly(int(val['freeCapacity'])),
+                         system_type=get_system_type(),
+                         )
+
+        key.incident_key = pddata['incident_key']
+        key.save()
+    else:
+        pass
+
+
+def storage_above_safelevel(key, val):
+    """
+    Called by the main task when a storage is above the minimum safe level
+    :param key: object from the incident_key model
+    :param val: metadata returned from storage_data map about the storage
+    :return: None
+    """
+    percent_free = return_percentage(int(val['capacity']), int(val['freeCapacity']))
+    if key.incident_key != '':
+        pddata = notify_pagerduty("Storage {0} no longer lacks sufficient free space. It is {1}% full.".format(val['contentDict']['name'],percent_free),
+                                  "resolve",
+                                  key.incident_key,
+                                  vidis_id=val['name'],
+                                  storage_name=val['contentDict']['name'],
+                                  free_capacity=human_friendly(int(val['freeCapacity'])),
+                                  system_type=get_system_type()
+                                  )
+        #if the HTTP request failed then we should have raised an exception
+        key.incident_key = ''
+        key.save()
+
+@periodic_task(run_every=(crontab(minute='*/10')), name="check_storage", ignore_result=True)
+def check_storage(celery_app=None):
+    """
+    Runs the main loop to check storages.  This is run as a periodic task by celery beat.
+    :param celery_app: pass in a fake celery app instance, for testing.  In production usage, leave this out and it defaults to None
+    :return: String indicating the system status, for celery flower
+    """
+    from gnmvidispine.vs_storage import VSStoragePathMap
+    from django.conf import settings
+    from models import StorageData, IncidentKeys
+    from time import sleep
+
+    log.info("check_storage run by Celery.")
+
+    if celery_app is None:
+        app = Celery()
+    else:
+        app=celery_app
+
+    i = app.control.inspect()
+
+    running = str(i.active())
+
+    rnumber = running.count("check_storage")
+
+    if rnumber > 1:
+        return "Task aborted to avoid duplication"
```

Comment:
Make this a raise RuntimeError("Task aborted.... so that the task will show up as failed